### PR TITLE
multipath-tools: 0.8.3 -> 0.9.3

### PR DESCRIPTION
**Pull request by risicle**
###### Description of changes

https://nvd.nist.gov/vuln/detail/CVE-2022-41973
https://nvd.nist.gov/vuln/detail/CVE-2022-41974

Enable tests & also so include the detail that this is to do with multipath _storage_ in `meta.description` - I had it in my mind that this was about multipath tcp...

:point_down: using ofborg for testing today.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by risicle** [comment]
`multipath-tools.passthru.tests` looks like it's already broken.

File: pkgs/os-specific/linux/multipath-tools/default.nix [hunk 1]
```diff
@@ -1,27 +1,30 @@
...
+  src = fetchFromGitHub {
+    owner = "opensvc";
+    repo = "multipath-tools";
+    rev = "refs/tags/${version}";
+    sha256 = "sha256-pIGeZ+jB+6GqkfVN83axHIuY/BobQ+zs+tH+MkLIln0=";
   };
 
-  patches = [
-    # fix build with json-c 0.14 https://www.redhat.com/archives/dm-devel/2020-May/msg00261.html
-    ./json-c-0.14.patch
```
Comment (LeSuisse) [inline pkgs/os-specific/linux/multipath-tools/default.nix:15]: Is the patch used somewhere else or can we remove the file?

**Review by LeSuisse — CHANGES_REQUESTED** [review]
Change looks good overall and it builds. Binaries seems to execute just fine too. I do not really have a setup at hand to do more tests.

File: pkgs/os-specific/linux/multipath-tools/default.nix (hunk 1 above)
Comment (risicle) [reply pkgs/os-specific/linux/multipath-tools/default.nix:15]: Well caught, totally forgot to delete it.

**Review by LeSuisse — APPROVED** [review]

**Comment by ajs124** [comment]
I think this broke the test/module, because `multipath_dir` was dropped.

**Comment by risicle** [comment]
So this _double_-broke the nixos test on unstable? Hard to avoid really...

**Comment by ajs124** [comment]
It should have been working prior to this. It passes with #198166 and a39a869cac6d40c301affca3b54b3b48ea2a1b19 reverted for me.

I'm obviously not advocating for reverting this, just did it locally for testing.

**Pull request merged**